codesign/inputparse.go

```diff
@@ -0,0 +1,110 @@
+package codesign
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bitrise-io/go-steputils/stepconf"
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/sliceutil"
+	"github.com/bitrise-io/go-xcode/autocodesign"
+	"github.com/bitrise-io/go-xcode/autocodesign/certdownloader"
+	"github.com/bitrise-io/go-xcode/autocodesign/keychain"
+)
+
+// StepInputParser ...
+type StepInputParser struct {
+	DistributionMethod        string
+	CertificateURLList        string
+	CertificatePassphraseList stepconf.Secret
+	KeychainPath              string
+	KeychainPassword          stepconf.Secret
+
+	CommandFactory command.Factory
+}
+
+// ParsedConfig ...
+type ParsedConfig struct {
+	CertificatesAndPassphrases []certdownloader.CertificateAndPassphrase
+	Keychain                   keychain.Keychain
+	DistributionMethod         autocodesign.DistributionType
+}
+
+// Parse validates and parses step inputs related to code signing, and returns with a ParsedConfig
+func (p StepInputParser) Parse(authType AuthType) (ParsedConfig, error) {
+	var (
+		certificatesAndPassphrases []certdownloader.CertificateAndPassphrase
+		keychainWriter             keychain.Keychain
+	)
+
+	if authType != NoAuth {
+		if strings.TrimSpace(p.CertificateURLList) == "" {
+			return ParsedConfig{}, fmt.Errorf("Code signing certificate URL: required variable is not present")
+		}
+		if strings.TrimSpace(p.KeychainPath) == "" {
+			return ParsedConfig{}, fmt.Errorf("Keychain path: required variable is not present")
+		}
+		if strings.TrimSpace(string(p.KeychainPassword)) == "" {
+			return ParsedConfig{}, fmt.Errorf("Keychain password: required variable is not present")
+		}
+
+		var err error
+		certificatesAndPassphrases, err = p.ParseCertificates()
+		if err != nil {
+			return ParsedConfig{}, fmt.Errorf("failed to parse certificate URL and passphrase inputs: %s", err)
+		}
+
+		keychainP, err := keychain.New(p.KeychainPath, p.KeychainPassword, p.CommandFactory)
+		if err != nil {
+			return ParsedConfig{}, fmt.Errorf("failed to open keychain: %s", err)
+		}
+		keychainWriter = *keychainP
+	}
+
+	return ParsedConfig{
```

Comment:
is ParsedConfig valid without a CertificatesAndPassphrases and Keychain (in case of authType == NoAuth)?